tests/test_0107_html_node_compare.py

```diff
@@ -0,0 +1,78 @@
+def test_node_comparisons():
+    from lona.html import Widget, Span, Node, Div
+    from lona.html.text_node import TextNode
+
+    class TestWidget(Widget):
+        def __init__(self, *nodes, **widget_data):
+            self.nodes = nodes
+            self.data = widget_data
+
+    # simple comparison
+    assert Div() == Div()
+    assert Div() is not Div()
+    assert Div() != Widget()
+    assert Div() != TextNode('')
+
+    # text nodes
+    assert TextNode('foo') == TextNode('foo')
+    assert TextNode('foo') != TextNode('bar')
+    assert TextNode('foo') is not TextNode('foo')
+
+    # legacy widgets
+    # TODO: remove in 2.0
+    assert TestWidget() == TestWidget()
+
+    assert TestWidget(Div()) == TestWidget(Div())
+    assert TestWidget(Div()) != TestWidget(Span())
+
+    assert TestWidget(foo=1) == TestWidget(foo=1)
```

Comment:
I think that this test can be split into multiple ones.

Reply:
Good idea. Done!